airbyte-cdk/python/airbyte_cdk/utils/airbyte_secrets_utils.py

```diff
@@ -18,3 +18,19 @@ def get_secrets(source: Source, config: Mapping[str, Any], logger: logging.Logge
         ".".join(key.split(".")[:1]) for key, value in flattened_key_values.items() if value and key.endswith("airbyte_secret")
     ]
     return [str(get_value_by_dot_notation(config, key)) for key in secret_key_names if config.get(key)]
+
+
+class AirbyteSecretHelper:
+    _secrets: List[str] = []
+
+    @classmethod
+    def update_secrets(cls, secrets: List[str]):
+        """Update the list of secrets to be replaced"""
+        cls._secrets = secrets
+
+    @classmethod
+    def filter_secrets(cls, string: str) -> str:
+        """Filter secrets from a string by replacing them with ****"""
+        for secret in cls._secrets:
+            string = string.replace(secret, "****")
+        return string
```

Comment:
Moved out the logic from `AirbyteLogFormatter` so it can be re-used (using this for trace messages now as well)